share/availability/light/availability.go

```diff
@@ -123,19 +124,19 @@ func (la *ShareAvailability) SharesAvailable(ctx context.Context, header *header
 	}
 	wg.Wait()
 
-	if errors.Is(ctx.Err(), context.Canceled) {
-		// Availability did not complete due to context cancellation, return context error instead of
-		// share.ErrNotAvailable
-		return ctx.Err()
-	}
-
 	// store the result of the sampling session
 	bs := encodeSamples(failedSamples)
 	la.dsLk.Lock()
 	err = la.ds.Put(ctx, key, bs)
 	la.dsLk.Unlock()
 	if err != nil {
-		log.Errorw("Failed to store sampling result", "error", err)
+		return fmt.Errorf("failed to store sampling result: %w", err)
```

Comment:
Good question! Should be logged by DASer. Daser logging revised here https://github.com/celestiaorg/celestia-node/pull/3889

Reply:
Ok, would be good to still log the `log.Errorw("availability validation failed"` below, instead of short-circuiting DB error

Reply:
>log.Errorw("availability validation failed")

It is logged when samples were not obtained in time, resulting in share.ErrNotAvailable error, which is not related to unexpected errors like put/get failures. 

I don't think we need to log any in Availability because they are propagated up the stack and can be logged by the caller. Removing such error logs from Availability would remove potential double logging of same errors. 

Additionally, I would rather remove all remaining logging of propagated errors from Availability in favour of logging in DASer. But it is subject to https://github.com/celestiaorg/celestia-node/pull/3889